pkg/services/sqlstore/stats.go

```diff
@@ -185,6 +186,58 @@ func GetSystemUserCountStats(ctx context.Context, query *models.GetSystemUserCou
 	})
 }
 
+type memoConcurrentUserStats struct {
+	stats []*models.ConcurrentUsersStats
+
+	memoized time.Time
+}
+
+var (
+	concurrentUserStatsCache         = memoConcurrentUserStats{}
+	concurrentUserStatsCacheLifetime = time.Hour
+)
+
+func GetConcurrentUsersStats(ctx context.Context, query *models.GetConcurrentUsersStatsQuery) error {
+	err := updateConcurrentUsersStatsIfNecessary(ctx, query.MustRefresh)
+	if err != nil {
+		return err
+	}
+	query.Result = concurrentUserStatsCache.stats
+	return nil
+}
+
+func updateConcurrentUsersStatsIfNecessary(ctx context.Context, refresh bool) error {
+	memoizationPeriod := time.Now().Add(-concurrentUserStatsCacheLifetime)
+	if refresh || concurrentUserStatsCache.memoized.Before(memoizationPeriod) {
+		err := updateConcurrentUsersStats(ctx)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func updateConcurrentUsersStats(ctx context.Context) error {
+	/*
+		Retrieves concurrent users stats as a histogram. For buckets, it rounds each token count data point up to the nearest multiple of 5
+		and then groups by that rounded value. If there is no data in a specific bucket, that row will not appear in the results.
+		Note that because ceil() is not a built-in function SQLite, rounding is used instead, which is sufficient approximation for this use case.
+	*/
+	var rawSql = `
+SELECT round(tokens/5.00 + 0.49999)*5	AS bucket_active_tokens,
```

Comment:
I thought distributing data across buckets with increment value of 5 is OK, but we can simply change the formula to the  `round(tokens/1.00 + 0.49999)*1` and have increments of 1.